karapace/protobuf/proto_normalizations.py

```diff
@@ -0,0 +1,282 @@
+"""
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+from karapace.dependency import Dependency
+from karapace.protobuf.enum_constant_element import EnumConstantElement
+from karapace.protobuf.enum_element import EnumElement
+from karapace.protobuf.extend_element import ExtendElement
+from karapace.protobuf.field_element import FieldElement
+from karapace.protobuf.group_element import GroupElement
+from karapace.protobuf.message_element import MessageElement
+from karapace.protobuf.one_of_element import OneOfElement
+from karapace.protobuf.option_element import OptionElement
+from karapace.protobuf.proto_file_element import ProtoFileElement
+from karapace.protobuf.rpc_element import RpcElement
+from karapace.protobuf.schema import ProtobufSchema
+from karapace.protobuf.service_element import ServiceElement
+from karapace.protobuf.type_element import TypeElement
+from karapace.schema_references import Reference
+from typing import List, Mapping, Optional, Sequence
+
+
+def sort_by_name(element: OptionElement) -> str:
+    return element.name
+
+
+class NormalizedRpcElement(RpcElement):
+    pass
+
+
+class NormalizedServiceElement(ServiceElement):
+    rpcs: Optional[Sequence[NormalizedRpcElement]] = None
+
+
+class NormalizedFieldElement(FieldElement):
+    pass
+
+
+class NormalizedExtendElement(ExtendElement):
+    fields: Optional[Sequence[NormalizedFieldElement]] = None
+
+
+class NormalizedTypeElement(TypeElement):
+    nested_types: Sequence["NormalizedTypeElement"]
+
+    def to_schema(self) -> str:
+        """Convert the object to valid protobuf syntax.
+
+        This must be implemented by subclasses.
+        """
+        raise NotImplementedError()
+
+
+class NormalizedProtoFileElement(ProtoFileElement):
+    types: Sequence[NormalizedTypeElement]
+    services: Sequence[NormalizedServiceElement]
+    extend_declarations: Sequence[NormalizedExtendElement]
+
+
+class NormalizedMessageElement(NormalizedTypeElement, MessageElement):
+    nested_types: Sequence[NormalizedTypeElement]
+    fields: Sequence[NormalizedFieldElement]
+    one_ofs: Sequence[OneOfElement]
+    groups: Sequence[GroupElement]
+
+    def to_schema(self) -> str:
+        return MessageElement.to_schema(self)
```

Comment:
```suggestion
class NormalizedMessageElement(MessageElement, NormalizedTypeElement):
    nested_types: Sequence[NormalizedTypeElement]
    fields: Sequence[NormalizedFieldElement]
    one_ofs: Sequence[OneOfElement]
    groups: Sequence[GroupElement]
```

issue: This looks like wrong MRO.

Reply:
ah you are right, thanks! Multi inheritance its complex :P

Reply:
same its true also for `NormalizedEnumElement`

Reply:
Unmarked as resolved as the point here was also that you should not need to define `to_schema` when the MRO is correct.